crates/gen-rust-lib/src/lib.rs

```diff
@@ -492,11 +492,22 @@ pub trait RustGenerator {
         id: TypeId,
         record: &Record,
         docs: &Docs,
+        derive_component: bool,
     ) {
         let info = self.info(id);
         for (name, mode) in self.modes_of(iface, id) {
             let lt = self.lifetime_for(&info, mode);
             self.rustdoc(docs);
+
+            if derive_component {
+                self.push_str("#[derive(wasmtime::component::ComponentType)]\n");
+                if self.lifetime_for(&info, mode).is_none() {
```

Comment:
I think `lt` can be used from above instead of calling `lifetime_for` again?

Reply:
thanks!